webapp/javascript/components/TimelineChart/TimelineTooltip.tsx

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable 
```

Comment:
Did you write this file from scratch? Is it forked from somewhere else?

Reply:
I wrote this file from scratch, here is main logic of new tooltip.